builtin/credential/oidc/backend.go

```diff
@@ -0,0 +1,48 @@
+package oidc
+
+import (
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func Factory(conf *logical.BackendConfig) (logical.Backend, error) {
+	return Backend().Setup(conf)
+}
+
+func Backend() *backend {
+	var b backend
```

Comment:
I'd recommend spawning a goroutine here to start trying to grab the provider and verifier structs. Later calls can then avoid the overhead of creating them themselves. Cancel any external work when backend.Cleanup is called.

Does that sound reasonable or is there no precedence for this?